### ISSUE TITLE
New analyzer: Enum types on left and right side of the 'HasFlag' should be the same

### ISSUE THREAD

**Issue by Evangelink** [dotnet/roslyn-analyzers#3529]
Add an analyzer that will report a diagnostic on a call to the `HasFlag` method for which the enum type on which it is invoked is not the same as the one passed as argument.

```csharp
public class C
{
	[Flags]
	public enum MyEnum { A, B, }
	[Flags]
	public enum OtherEnum { A, }

	public void Do(MyEnum m)
	{
		m.HasFlag(OtherEnum.A); // Report here
	}
}
```

**Comment by sharwell** [comment]
I think this would be a valuable rule.

**Comment by mavasani** [comment]
Sounds good @Evangelink
